docs/docs/llms/llm-flow-classifier.mdx

```diff
@@ -11,3 +11,145 @@ import RasaDiscoveryBanner from "@theme/RasaDiscoveryBanner";
+
+Here is what happened previously in the conversation:
+USER: I want to send some money to Joe
+AI: How much money do you want to transfer?
+USER: 40$
+
+
+You are currently in the flow "transfer_money".
+You have just asked the user for the slot "transfer_money_amount_of_money".
+
+
+Here are the slots of the currently active flow with their names and values:
+
+- transfer_money_recipient: Joe
+
+- transfer_money_amount_of_money: undefined
+
+- transfer_money_final_confirmation: undefined
+
+
+
+If you start a flow, you can already fill that flow's slots with information the user provided for starting the flow.
+
+The user just said """40$""".
+
+Based on this information generate a list of actions you want to take. Your job is to start flows and to fill slots where appropriate. Any logic of what happens afterwards is handled by the flow engine. These are your available actions:
+* Slot setting, described by "SetSlot(slot_name, slot_value)". An example would be "SetSlot(recipient, Freddy)"
+* Starting another flow, described by "StartFlow(flow_name)". An example would be "StartFlow(transfer_money)"
+* Cancelling the current flow, describe by "CancelFlow()"
+
+Write out the actions you want to take for the last user message, one per line.
+Do not prematurely fill slots with abstract values.
+Only use information provided by the user.
+Strictly adhere to the provided action types above for starting flows and setting slots.
+Focus on the last message and take it one step at a time.
+Use the previous conversation steps only to aid understanding.
+The action list:
+```
+
+The LLM might reply to this prompt with:
+```
+SetSlot(transfer_money_amount_of_money, 40$)
+```
+Which we would then turn into an `inform` intent, since no other flow needs to be
+started and the information belongs to the question that was just asked. Also,
+the `transfer_money_amount_of_money` entity would be present.
+
+If the user had said "Sorry, I meant to John" instead, the LLM might have replied with
+```
+SetSlot(transfer_money_recipient, John)
+```
+Which we then would turn into an `correction` intent, since it tries to set a slot that
+was asked earlier in the flow. Also, the `transfer_money_recipient` entity would be
+present.
+
+If the user had said "How much do I still have in my account" instead, the LLM might
+have replied with
+```
+StartFlow(check_balance)
+```
+Which we then would turn into a `check_balance` intent.
+
+It is also possible to return multiple actions at the same time for the LLM, if the
+user for example says "I want to send 40$ to John", the LLM should reply:
+```
+StartFlow(transfer_money)
+SetSlot(transfer_money_recipient, John)
+SetSlot(transfer_money_amount_of_money, 40$)
+```
+
+which would then be turned into the `transfer_money` intent as well as two entities.
+
+Certain asks by the user and responses by the LLM are currently deemed as too complex
+and classified as such based on the composition of the action list.
```

Comment:
can we explain a bit more about why we needed to do this? and can you explain what happens when something is classified as "too complex": is the result that the output of the classifier is ignored?

Reply:
Sure thing, users can technically come up with arbitrarily complex commands like "start 1000000 money transfers" but on a more benign site they might just try to do a lot of things at once "50$ to James. And then please tell me how much money I still have left, and my recent transactions". For now, we just say, let's take it step by step. Doesn't mean we can't allow for more complexity later. We might actually be able it, but I think it's also good if there's some limit and we communicate that instead of attempting and failing.